src/app/cli/src/testing/kamu.rs

```diff
@@ -333,6 +326,37 @@ impl Kamu {
 
         (pull_aliases, push_aliases)
     }
+
+    async fn execute_impl<I, S, T>(
+        &self,
+        cmd: I,
+        maybe_input: Option<T>,
+    ) -> assert_cmd::assert::Assert
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<ffi::OsStr>,
+        T: Into<Vec<u8>>,
+    {
+        let mut command = assert_cmd::Command::cargo_bin("kamu-cli").unwrap();
+
+        command.current_dir(self.workspace_path.clone());
+
+        if let Some(system_time) = &self.system_time {
+            let system_time = system_time.to_rfc3339_opts(chrono::SecondsFormat::Millis, true);
+
+            command.args(["--system-time", system_time.as_str()]);
+        }
+
+        command.args(cmd);
+
+        if let Some(input) = maybe_input {
+            command.write_stdin(input);
+        }
+
+        tokio::task::spawn_blocking(move || command.assert())
```

Comment:
I'm confused. I thought the point of this change was to avoid linking `kamu-cli` in all e2e tests, to speed up build times, but here I see:
- the testing wrapper code itself is in `kamu-cli` library - so it has to be a dependency and be linked
- a lot of functions in this test wrapper (`add_dataset`, `get_list_of_repo_aliases`...) are still using in-process calls - so I don't know how smart the linkers are about throwing away code that isn't used but one call to `add_dataset` from e2e test will probably immediately result in a bunch of extra dependencies being brought in

Reply:
Fixed